File: files/CSEC 16022025.py
import pandas as pd

path = "files/Ex-Challenge 07 2025.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=2, nrows=11)
test = pd.read_excel(path, usecols="D", skiprows=2, nrows=15).astype(str)

input['group'] = (input['Staff'].diff() < 0).cumsum() + 1
input['Staff'] = input['Staff'].astype(str)

result = input.groupby('group', group_keys=False).apply(
    lambda x: x._append(pd.DataFrame({'Staff': [f"GROUP {x.name}"]}))
).reset_index(drop=True)[['Staff']].rename(columns={'Staff': 'Groups'})

print(result.equals(test))

File: files/2025-02-16/CSEC 16022025.R
library(tidyverse)
library(readxl)

path = "files/Ex-Challenge 07 2025.xlsx"
input = read_excel(path, range = "B3:B14")
test  = read_excel(path, range = "D3:D18")

result = input %>%
  mutate(group = cumsum(c(1, diff(Staff) < 0)),
         Staff  = as.character(Staff)) %>%
  group_by(group) %>%
  group_split() %>%
  imap_dfr(~ {
    dynamic_row <- tibble(
      Staff = paste("GROUP", .y),
      group = unique(.x$group)
    )
    bind_rows(.x, dynamic_row)
  }) %>%
  select(Groups = Staff)

all.equal(result, test)
# [1] TRUE